code/modules/mob/living/simple_animal/hostile/grue.dm

```diff
@@ -0,0 +1,591 @@
+		icon_state = "gruespawn_living"
+		icon_living = "gruespawn_living"
+		icon_dead = "gruespawn_dead"
+		melee_damage_lower = 1
+		melee_damage_upper = 5
+		attacktext = "bites"
+		maxHealth=50
+		moultcost=250
+		burnmalus=1
+		environment_smash_flags = 0
+		attack_sound = 'sound/weapons/bite.ogg'
+		size = SIZE_SMALL
+		pass_flags = PASSTABLE
+	else if (lifestage==2)
+		name = "grue"
+		desc = "A creeping thing that lives in the dark. It is still a juvenile."
+		icon_state = "grueling_living"
+		icon_living = "grueling_living"
+		icon_dead = "grueling_dead"
+		melee_damage_lower = 10
+		melee_damage_upper = 15
+		attacktext = "chomps"
+		maxHealth=100
+		moultcost=500
+		burnmalus=2**(1/3)
+		environment_smash_flags = SMASH_LIGHT_STRUCTURES | SMASH_CONTAINERS | OPEN_DOOR_WEAK | OPEN_DOOR_STRONG
+		attack_sound = 'sound/weapons/cbar_hitbod1.ogg'
+		size = SIZE_NORMAL
+		pass_flags = 0
+		force_airlock_time=100
+	else if (lifestage>=3)
+		name = "grue"
+		desc = "A dangerous thing that lives in the dark."
+		icon_state = "grue_living"
+		icon_living = "grue_living"
+		icon_dead = "grue_dead"
+		attacktext = "gnashes"
+		maxHealth = 200
+		moultcost=0 //not needed for adults
+		melee_damage_lower = 20
+		melee_damage_upper = 30
+		melee_damage_type = BRUTE
+		held_items = list()
+		burnmalus=3**(1/3)
+		environment_smash_flags = SMASH_LIGHT_STRUCTURES | SMASH_CONTAINERS | OPEN_DOOR_WEAK | OPEN_DOOR_STRONG
+		attack_sound = 'sound/weapons/cbar_hitbod1.ogg'
+		size = SIZE_BIG
+		pass_flags = 0
+		force_airlock_time=0
+	health=tempHealth*maxHealth
+
+//Grue vision
+/mob/living/simple_animal/hostile/grue/update_perception()
+
+	if(client)
+		if(client.darkness_planemaster&&a_use_alpha)
+			client.darkness_planemaster.blend_mode = BLEND_ADD
+			client.darkness_planemaster.alpha = a_blend_add_test
+		client.color = list(
+					1,0,0,1,
+					-1,0.2,0.2,1,
+	 				-1,0.2,0.2,1,
+		 			0,0,0,1,
+		 			0,0,0,0)
+
+		if(a_matrix_testing_override) //not used for now
+			client.color = list(a_11,a_12,a_13,a_14,
+								a_21,a_22,a_23,a_24,
+		 						a_31,a_32,a_33,a_34,
+			 					a_41,a_42,a_43,a_44,
+			 					a_51,a_52,a_53,a_54)
+//		else if(b_matrix_testing_override) //not used for now
+//			colourmatrix = list(b_11, b_12, b_22,
+//						 b_21,b_22, b_23,
+//						 b_31, b_32, b_33,
+//						 b_41, b_42, b_43)
+
+/mob/living/simple_animal/hostile/grue/Stat()
+	..()
+	if(statpanel("Status"))
+//		stat(null, "Intent: [a_intent]")
+//		stat(null, "Move Mode: [m_intent]")
+		if(ismoulting)
+			stat(null, "Moulting progress: [round(100*(1-moulttimer/moulttime),0.1)]%")
+		if(lifestage<3) //not needed for adults
+			stat(null, "Shadow power: [round(shadowpower,0.1)]/[round(maxshadowpower,0.1)]")
+		if(lifestage>=3)
+			if(config.grue_egglaying)
+				stat(null, "Reproductive energy: [eatencharge]")
+			stat(null, "Sentient life forms eaten: [eatencount]")
+
+/mob/living/simple_animal/hostile/grue/gruespawn
```

Comment:
Remove the var overrides you did previously and put them here.

Reply:
I'm not sure what you mean by this. Could you please clarify?

Reply:
The variables change as the grue moults into a new lifestage, so by having them changed by the lifestage_updates() proc this lets the variables be set by both New() and by moulting. If I changed it from that, then they wouldn't update properly after moulting.